reporter/src/main/kotlin/StatisticsCalculator.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.model.utils
+
+import com.here.ort.model.DependencyTreeStatistics
+import com.here.ort.model.IssueStatistics
+import com.here.ort.model.OrtResult
+import com.here.ort.model.Severity
+import com.here.ort.model.Statistics
+import com.here.ort.reporter.ResolutionProvider
+
+/**
+ * This class calculates [Statistics] for a given [OrtResult] and applicable
+ * [com.here.ort.model.config.ErrorResolution]s and applicable [com.here.ort.model.config.RuleViolationResolution]s.
+ */
+internal class StatisticsCalculator {
+    /**
+     * Return the [Statistics] for the given [ortResult].
+     */
+    fun getStatistics(ortResult: OrtResult, resolutionProvider: ResolutionProvider) = Statistics(
+        openIssues = getOpenIssuesIssues(ortResult, resolutionProvider),
+        openRuleViolations = getOpenRuleViolations(ortResult, resolutionProvider),
+        dependencyTree = DependencyTreeStatistics(
+            includedProjects = ortResult.getProjects().count { !ortResult.isExcluded(it.id) },
+            excludedProjects = ortResult.getProjects().count { ortResult.isExcluded(it.id) },
```

Comment:
Can't we avoid counting twice here by basically doing `ortResult.getProjects().size - includedProjects`? Similar below.

Reply:
It can be avoided, sure. I prefer to keep it as is since I find the code more readable.

Reply:
Given that you were putting quite some effort into fixing performance issues (when evaluating rules, IIRC), I'm a bit surprised by this, I have to admit.

Reply:
I'm actually surprised you seem not to be aware that I always prefer optimizing for readability over optimizing for performance if performance gains are unimportant.

Reply:
I'm actually surprised you seem not to be aware that our views of what is readable or not differ quite a bit. To me `ortResult.getProjects().size - includedProjects` is not less readable. But if you say performance does not matter here (and never will), I take your word for it.

Reply:
I'm not saying it never will. I think it is fine to defer performance optimization until they turn out to be needed. 

Your initial point was `performance` not `readability`. If you now wanna argue that your proposed code is more readable than go for it, otherwise I'm fine keeping things as is.

Reply:
> I think it is fine to defer performance optimization until they turn out to be needed.

At the cost of profiling where exactly the performance is lost then again...

> If you now wanna argue that your proposed code is more readable than go for it

I wrote "is not less readable", and that's what I meant.

Reply:
I still prefer to keep as is, same for the below. As we disagree propose to resolve this by you telling me whether this is a `must fix` or `optional`. I'll fix it if it's the former.

Reply:
It's optional, which is why I only posted it as a comment, and not requested changes to this PR. That's my usual way of expressing `must fix` vs. `optional`.

Reply:
Ok, thanks. Resolving this discussion then.